src/MojoMediaObjectSerializer.cc

```diff
@@ -28,34 +28,34 @@ void MojoMediaObjectSerializer::SerializeToDatabaseObject(const MediaFile& file,
     MojErr err;
 
     err = obj.putString("path", file.path().c_str());
-    //We only need name and modifiedTime for certain types.
-	//err = obj.putString("name", file.name().c_str());
-    //err = obj.putInt("modifiedTime", file.modifiedTime());
-    err = obj.putInt("createdTime", file.createdTime());
     err = obj.putInt("size", file.size());
-    //err = obj.putString("searchKey", file.searchKey().c_str());
 
     if (file.type() == MediaType::AudioMedia) {
         err = obj.putString("_kind", "com.palm.media.audio.file:1");
 
         err = obj.putString("albumArtist", file.albumArtist().c_str());
         err = obj.putString("album", file.album().c_str());
-		err = obj.putInt("modifiedTime", file.modifiedTime());
-
+        err = obj.putInt("modifiedTime", file.modifiedTime());
+        err = obj.putInt("createdTime", file.createdTime());
+     
         MojObject trackObj(MojObject::Type::TypeObject);
         trackObj.putInt("position", file.trackPosition());
         trackObj.putInt("total", file.trackTotal());
         err = obj.put("track", trackObj);
 
         MojObject discObj(MojObject::Type::TypeObject);
-        //FIXME, this should return 1 and 1 when nothing available.
-		//trackObj.putInt("position", file.discPosition());
-        //trackObj.putInt("total", file.discTotal());
-        trackObj.putInt("position", "1");
-        trackObj.putInt("total", "1");
-
+        discObj.putInt("position", file.discPosition());
+        discObj.putInt("total", file.discTotal());
         err = obj.put("disc", discObj);
 
+        MojObject sortKeyObj(MojObject::Type::TypeObject);
+        sortKeyObj.putString("albumArtistDiscAndTrack", (file.album()+"\t\t"+file.artist()+"\t\t"+file.album()+"\t\t"+file.discPosition()*100000+file.trackPosition()).c_str());
+        sortKeyObj.putString("albumDiscAndTrack", (file.album()+"\t\t"+file.discPosition()*100000+file.trackPosition()).c_str());
```

Comment:
In C++11, there is "std::to_string(my_integer);", which could here be used like this:
std::string strDiscPos = std::to_string(file.discPosition()*100000+file.trackPosition());
sortKeyObj.putString("albumDiscAndTrack", (file.album()+"\t\t"+strDiscPos).c_str();
Could be done in one long line, but wouldn't be that nice to read...